app/assets/javascripts/controllers/generic_object/generic_object_definition_controller.js

```diff
@@ -0,0 +1,79 @@
+ManageIQ.angular.app.controller('genericObjectDefinitionFormController', ['$http', '$scope', 'miqService', 'genericObjectSubscriptionService', function($http, $scope, miqService, genericObjectSubscriptionService) {
+  var init = function() {
+    hideAndClearForm();
+
+    $scope.newRecord = true;
+    $scope.showSingleItem = false;
+
+    genericObjectSubscriptionService.subscribeToShowAddForm(showAddForm);
+    genericObjectSubscriptionService.subscribeToTreeClicks(showSelectedItem);
+    genericObjectSubscriptionService.subscribeToRootTreeclicks(showAllItems);
+  };
+
+  var showAddForm = function(_response) {
+    $scope.clearForm();
+    $scope.showAddForm = true;
+    $scope.$apply();
+    sendDataWithRx({eventType: 'deselectTreeNodes'});
+  };
+
+  var showAllItems = function(response) {
+    $scope.genericObjectList = response;
+    $scope.showSingleItem = false;
+    $scope.$apply();
+  };
+
+  var addedGenericObject = function(data) {
+    var successCallback = function(response) {
+      sendDataWithRx({eventType: 'treeUpdated', response: JSON.parse(response.data.tree_data)});
+    };
+
+    $http.get('tree_data').then(successCallback);
+  };
+
+  var hideAndClearForm = function() {
+    $scope.clearForm();
+    $scope.showAddForm = false;
+  };
+
+  var showSelectedItem = function(response) {
+    var modelData = response;
```

Comment:
Yeah, no reason, it's a remnant from when `response` had `response.data` on it. Will fix.